aws_eden_cli/cmdline.py

```diff
@@ -116,6 +123,91 @@ def command_config(args: dict):
         else:
             logger.info(f"Found {errors} errors")
 
+    elif args['push']:
+        dynamodb = boto3.client('dynamodb')
+
+        table_name = "eden"
+
+        try:
+            response = dynamodb.describe_table(TableName=table_name)
+            table_status = response['Table']['TableStatus']
+        except botocore.exceptions.NoCredentialsError:
+            logger.error("AWS credentials not found!")
+            return
+        except Exception as e:
+            if hasattr(e, 'response') and 'Error' in e.response:
+                if e.response['Error']['Code'] != 'ResourceNotFoundException':
+                    logger.error(f"Unknown exception raised on DescribeTable API: {e.response}")
+                    return
+            else:
+                logger.error(f"Unknown exception raised on DescribeTable API: {e}")
+                return
+
+            response = dynamodb.create_table(
+                AttributeDefinitions=[
+                    {
+                        'AttributeName': 'env_name',
+                        'AttributeType': 'S'
+                    },
+                    {
+                        'AttributeName': 'last_updated',
+                        'AttributeType': 'S'
+                    }
+                ],
+                TableName=table_name,
+                KeySchema=[
+                    {
+                        'AttributeName': 'env_name',
+                        'KeyType': 'HASH'
+                    },
+                ],
+                GlobalSecondaryIndexes=[
+                    {
+                        'IndexName': 'env_name_last_updated_gsi',
+                        'KeySchema': [
+                            {
+                                'AttributeName': 'env_name',
+                                'KeyType': 'HASH',
+                            },
+                            {
+                                'AttributeName': 'last_updated',
+                                'KeyType': 'RANGE',
+                            },
+                        ],
+                        'Projection': {
+                            'ProjectionType': 'ALL',
+                        }
+                    },
+                ],
+                BillingMode='PAY_PER_REQUEST',
+            )
+
+            table_status = response['TableDescription']['TableStatus']
+
+        if table_status == 'DELETING':
+            logger.error("Table deletion is in progress, try again later")
+            return
+
+        elif table_status == 'UPDATING':
```

Comment:
Should we wait for the completion of table update instead of returning an error immediately as we did in creating state?

Reply:
UPDATING state only happens when there are DB schema changes, GSI addition etc. that can be performed in-place. The table in question is for metadata only and is not expected to change once we ship v0.2. When there is a DB migration, I will think about waiting here too.

Reply:
Thanks! That makes sense.